packages/next-on-pages/templates/_worker.js/routes-matcher.ts

```diff
@@ -316,17 +316,22 @@ export class RoutesMatcher {
 		// NOTE: Special handling for `/index` RSC routes. Sometimes the Vercel build output config
 		// has a record to rewrite `^/` to `/index.rsc`, however, this will hit requests to pages
 		// that aren't `/`. In this case, we should check that the previous path is `/`.
-		if (/\/index\.rsc$/i.test(this.path) && !/^\/(?:index)?$/i.test(prevPath)) {
+		const isRscIndex = /\/index\.rsc$/i.test(this.path);
+		const isPrevAbsoluteIndex = /^\/(?:index)?$/i.test(prevPath);
+		const isPrevPrefetchRscIndex = /^\/__index\.prefetch\.rsc$/i.test(prevPath);
```

Comment:
a comment (and source code link) to this `__index.preferch.rsc` would be great :slightly_smiling_face: